2I backpressure

kellymclaughlin opened the pull request:
Use the same mechanism used for key listing to provide backpressure for secondary index requests.

kellymclaughlin committed:
Add backpressure to secondary index requests.

Fixes: #399

Use the same mechanism used for key listing to provide
backpressure for secondary index requests.

beerriot commented:
I think we'll have to do some capability negotiation and handling of both varieties of messages, or mixed-version clusters won't work.

kellymclaughlin commented:
Quite right. I will add that in.

beerriot committed:
negotiate whether or not to use index backpressure

this is so mixed-version clusters can understand each other's index
requests and responses

beerriot commented:
I think that should do it. Tested with a mixed cluster, one HEAD node and a 1.2 node, and also with two HEAD nodes. Have a look, @kellymclaughlin?
